asab/application.py

```diff
@@ -584,3 +589,32 @@ def time(self):
 		Return UTC unix timestamp using a loop time (a fast way how to get a wall clock time).
 		'''
 		return self.BaseTime + self.Loop.time()
+
+
+	# Housekeeping
+
+	def set_housekeeping_time_from_config(self):
+		"""Set the housekeeping time from `Config['general']['housekeeping_time']`
+		for today or tomorrow.
+		"""
+		config_house_time = datetime.datetime.strptime(Config['general']['housekeeping_time'], "%H:%M")  # default: 03:00
+		now = datetime.datetime.now(datetime.timezone.utc)
+		td_midnight = now - datetime.timedelta(
+			hours=now.hour,
+			minutes=now.minute,
+			seconds=now.second,
+			microseconds=now.microsecond)  # today at 00:00
+		next_housekeeping_time = td_midnight + datetime.timedelta(
+			hours=config_house_time.hour,
+			minutes=config_house_time.minute)  # today at the time for housekeeping
+		if now > next_housekeeping_time:
+			next_housekeeping_time += datetime.timedelta(days=1)
+		return next_housekeeping_time
+
+	def check_for_housekeeping(self, message_type):
+		"""Check if it's time for 'Application.housekeeping!'.
+		If so, publish the message and set housekeeping time for the next day.
+		"""
+		if datetime.datetime.now(datetime.timezone.utc) > self.NextHousekeeping:
+			self.PubSub.publish("Application.housekeeping!")
```

Comment:
- [ ] Comment this please
- [ ] Also ensure that the housekeeping is not executed immediately after app is started (after 3:00). The app has to wait till a first 3:00 ... or another check must be done that the current time is >3:00 AND <3:20 (for example).

Reply:
The second one should be already checked in `set_housekeeping_time_from_config()`, see the lines 609-610.
```python
if now > next_housekeeping_time:
	next_housekeeping_time += datetime.timedelta(days=1)
```